src/ppx/xr_component.cpp

```diff
@@ -0,0 +1,395 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if defined(PPX_BUILD_XR)
+#include "ppx/xr_component.h"
+#include "ppx/grfx/grfx_instance.h"
+#include <glm/gtc/type_ptr.hpp>
+
+namespace ppx {
+void XrComponent::InitializeBeforeGrfxDeviceInit(bool enableDebug)
+{
+    const char* graphicsAPIExtension = "";
+    // Extensions (Non-Optional)
+#if defined(PPX_D3D11)
+    graphicsAPIExtension = XR_KHR_D3D11_ENABLE_EXTENSION_NAME;
+#endif // defined(PPX_D3D11)
+
+#if defined(PPX_D3D12)
+    graphicsAPIExtension = XR_KHR_D3D12_ENABLE_EXTENSION_NAME;
+#endif // defined(PPX_D3D12)
+
+#if defined(PPX_VULKAN)
+    graphicsAPIExtension = XR_KHR_VULKAN_ENABLE2_EXTENSION_NAME;
+#endif // defined(PPX_VULKAN)
```

Comment:
This should not be based (only) on the static `#define`. BigWheels allows you to choose the API at runtime. The chosen API should be passed here so that the extension is chosen correctly based on that.

Reply:
fixed